pstatus-notifications-workflow-ktor/src/main/kotlin/workflow/NotificationWorkflowImpl.kt

```diff
@@ -1,18 +1,30 @@
 package gov.cdc.ocio.processingnotifications.workflow
 
+import com.azure.cosmos.models.CosmosQueryRequestOptions
 import gov.cdc.ocio.processingnotifications.activity.NotificationActivities
+import gov.cdc.ocio.processingnotifications.cosmos.CosmosRepository
+import gov.cdc.ocio.processingnotifications.utils.SqlClauseBuilder
 import io.temporal.activity.ActivityOptions
 import io.temporal.common.RetryOptions
 import io.temporal.workflow.Workflow
 import mu.KotlinLogging
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
 import java.time.Duration
+import java.time.LocalDate
+import java.time.ZoneId
+import java.time.ZoneOffset
+import java.time.format.DateTimeFormatter
 
 /**
  * The implementation class for notifying if an upload has not occurred within a specified time
  * @property activities T
  */
-class NotificationWorkflowImpl : NotificationWorkflow {
+class NotificationWorkflowImpl : NotificationWorkflow, KoinComponent {
     private val logger = KotlinLogging.logger {}
+
+    private val cosmosRepository by inject<CosmosRepository>()
+
     private val activities = Workflow.newActivityStub(
         NotificationActivities::class.java,
         ActivityOptions.newBuilder()
```

Comment:
The activity stub is not complete ... can you add the below part as well
  private val activities = Workflow.newActivityStub(
        NotificationActivities::class.java,
        ActivityOptions.newBuilder()
            .setStartToCloseTimeout(Duration.ofSeconds(10)) // Set the start-to-close timeout
            .setScheduleToCloseTimeout(Duration.ofMinutes(1)) // Set the schedule-to-close timeout
            .setRetryOptions(
                RetryOptions.newBuilder()
                    .setMaximumAttempts(3) // Set retry options if needed
                    .build()
            )
            .build()
    )

Reply:
This adds the retry policy and a start and schedule timeouts which when surpassed would retry